torch/lib/c10d/Utils.cpp

```diff
@@ -15,8 +16,10 @@
 #include <memory>
 #include <string>
 #include <thread>
+#endif
 
 namespace c10d {
+#ifndef _WIN32
```

Comment:
any reason for breaking ifndef into two segments and leave an empty `c10d` namespace in this file?

Reply:
Does it work if we wrap everything except `#include <c10d/Utils.hpp>` in one `#ifndef` block?